cbor/record.go

```diff
@@ -0,0 +1,217 @@
+package cbor
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/ipfs/go-cid"
+	"github.com/ipfs/go-ipld-cbor"
+	"github.com/ipfs/go-ipld-format"
+	ic "github.com/libp2p/go-libp2p-core/crypto"
+	mh "github.com/multiformats/go-multihash"
+	"github.com/textileio/go-textile-core/crypto"
+	"github.com/textileio/go-textile-core/thread"
+	pb "github.com/textileio/go-textile-threads/pb"
+)
+
+func init() {
+	cbornode.RegisterCborType(record{})
+}
+
+// record defines the node structure of a record.
+type record struct {
+	Block cid.Cid
+	Sig   []byte
+	Prev  cid.Cid `refmt:",omitempty"`
+}
+
+// NewRecord returns a new record from the given block and log private key.
+func NewRecord(ctx context.Context, dag format.DAGService, block format.Node, prev cid.Cid, sk ic.PrivKey, key crypto.EncryptionKey) (thread.Record, error) {
+	payload := block.RawData()
+	if prev.Defined() {
+		payload = append(payload, prev.Bytes()...)
+	}
+	sig, err := sk.Sign(payload)
```

Comment:
Hmm, yeah. I suppose if you're not using `Event`s which do not have any data (only links), this could get nasty. `prev` is a `Cid`, so `Bytes` is just returns the byte representation of the Cid. How about `block.Cid().Bytes()+prev.Bytes()`?

Reply:
LGTM